crates/block-producer/src/server.rs

```diff
@@ -57,6 +57,8 @@ impl BlockProducer {
         store_address: SocketAddr,
         batch_prover: Option<Url>,
         block_prover: Option<Url>,
+        batch_interval: Duration,
+        block_interval: Duration,
```

Comment:
Not for this PR, but we may need to create some kind of struct(s) for these. Otherwise, as we keep adding configuration options, we'll end up with quite a long list of parameters.

Reply:
Agreed. My plan/aim is that we can actually just use the `BlockProducer` as that struct. This would be possible once component startup is possible without this two-stage `init -> serve`.